src/graphs/SMG.cc

```diff
@@ -38,6 +39,16 @@ const std::map<SMGValue, SMGEdgePointsToPtr>& SMG::GetPTEdges() const { return p
 
 const SMGEntitySet<const SMGEdgeHasValue>& SMG::GetHVEdges() const { return hv_edges_; }
 
+const SMGEntitySet<const SMGEdgeHasValue> SMG::GetHVEdges(
+    const SMGEdgeHasValueFilter& filter) const {
+  SMGEntitySet<const SMGEdgeHasValue> to_return;
+  std::copy_if(hv_edges_.begin(),
+               hv_edges_.end(),
+               std::inserter(to_return.set(), to_return.begin()),
+               filter);
+  return to_return;
+}
```

Comment:
You're right, I didn't realize it, this is how it's implemented in original library. I personally prefer version with std::copy_if (I think the implementation is nicer - there shouldn't be difference in performance).

Moreover, FilterSet is not used anywhere besides tests, yet. I would reimplement it using copy_if. Maybe PM could tell if the different implementations were intentional?

----- Pôvodná správa -----
Od: "michkot" <notifications@github.com>
Odoslané: ‎16.‎3.‎2016 23:41
Komu: "VeriFIT/smg3" <smg3@noreply.github.com>
Kópia: "viktormalik" <viktor.malik@gmail.com>
Predmet: Re: [smg3] Finished porting of SMGConsistencyVerifier with tests(#13)

In src/graphs/SMG.cc:
> @@ -38,6 +39,16 @@ const std::map<SMGValue, SMGEdgePointsToPtr>& SMG::GetPTEdges() const { return p
>  
>  const SMGEntitySet<const SMGEdgeHasValue>& SMG::GetHVEdges() const { return hv_edges_; }
>  
> +const SMGEntitySet<const SMGEdgeHasValue> SMG::GetHVEdges(
> +    const SMGEdgeHasValueFilter& filter) const {
> +  SMGEntitySet<const SMGEdgeHasValue> to_return;
> +  std::copy_if(hv_edges_.begin(),
> +               hv_edges_.end(),
> +               std::inserter(to_return.set(), to_return.begin()),
> +               filter);
> +  return to_return;
> +}
That is obvious and not what I am addressing here. IMHO:
A) It should just call FilterSet(hv_edges_&), compiler might even optimize out the extra function call. 
B) aside from one function is using class field and the second function argument, they do the exactly same thing - but one with std::copy_if and the second with for-loop. Which is better? And - that should be unified.
—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub